packages/zowe-explorer/src/dataset/DatasetTree.ts

```diff
@@ -934,15 +934,15 @@ export class DatasetTree extends ZoweTreeProvider implements IZoweTree<IZoweData
                             placeHolder: localize("searchHistory.options.prompt", "Select a filter"),
                         };
                         // get user selection
-                        const choice = await Gui.showQuickPick([createPick, globals.SEPARATORS.RECENT_FILTERS, ...items], options1);
+                        const choice = await Gui.showQuickPick([createPick, globals.SEPARATORS.RECENT_FILTERS, ...items].filter(Boolean), options1);
```

Comment:
Could we add logic to `Gui.showQuickPick` to filter out the undefined items? This might reduce code duplication (except for places where we call `Gui.createQuickPick`).

Reply:
Since the return type can be `Thenable<string | vscode.QuickPickItem>`, I ran into some issues with type guards. I can investigate this a bit further but for now it might just be easier to have the filters for each list where they are needed.